samples/subsys/usb/mass/boards/nrf52840dk_nrf52840.overlay

```diff
@@ -18,3 +18,12 @@
 		};
 	};
 };
+
```

Comment:
There should also be mass storage configuration using internal NOR flash.

Reply:
Why? There wasn't such example before.

Reply:
OK, then I can add it. I have been asked about FAT FS configuration, which requires disk access, on internal NOR, so I guess it is sometimes used scenario. Exposing such FAT FS via USB is just extended scenario.

Reply:
I don't think it is wise unless the internal flash allows simultaneous code execution during erase (i.e. the code execution does not stop during erase). Otherwise you are likely to break USB constraints and risk the host simply dropping the device.

Reply:
> I don't think it is wise unless the internal flash allows simultaneous code execution during erase (i.e. the code execution does not stop during erase). Otherwise you are likely to break USB constraints and risk the host simply dropping the device.

Good point for the USB. I retreat here.

Reply:
@desowin, the flash_disk application is larger than USB. And if there is such a strict time constraint it might be problematic to allow a software update over usb using MSD (e.g. using uf2).

Reply:
Well, currently USB Mass Storage with flashdisk fatfs backend can exceed the host 30 second timeout on writes. All it takes is consecutive 203 sector write (203*512 bytes = 101.5 KiB file). The situation will likely get better when caching is implemented. For software update there is dedicated class that does not have the issue - USB DFU.

Reply:
> There should also be mass storage configuration using internal NOR flash.

No. 1) It would not be much different 2) We know it would be bad example

Reply:
Yes, the caching does not solve the problem but only delays the it. The USB specification itself does not specify any timeouts, in fact the USB Mass Storage is virtually a SCSI wrapper class. The timeouts are arbitrarily chosen by the host (i.e. fixed value, on Linux 30 seconds) and device has no way to change them.

The problem is that for both READ (10) or WRITE (10) it is the host that decides how many blocks to read. The only way for the device to limit it seems to be MAXIMUM TRANSFER LENGTH in Block Limits VPD page. Currently Zephyr Mass Storage class does not support the Block Limits VPD page. I don't know what would it take to support it (or if host would actually care about it) as my SCSI understanding is very limited.

Another issue, specifically related to use of internal flash as storage, is that if code execution is stalled during erase, then the device won't be able to respond to control transfers in time. Anyway, this is really outside the scope of this PR.

Reply:
I agree that this is outside of the scope of this PR.